core/src/main/java/com/homeaway/streamplatform/streamregistry/db/dao/impl/ConsumerDaoImpl.java

```diff
@@ -151,42 +155,53 @@ public void delete(String streamName, String actorName) {
         return Optional.of(AvroToJsonDTO.getJsonConsumer(consumer));
     }
 
-    private Optional<com.homeaway.streamplatform.streamregistry.model.Consumer> getConsumer(String streamName, String consumerName) {
+    private Optional<com.homeaway.streamplatform.streamregistry.model.Consumer> getConsumer(String streamName, String consumerName) throws StreamNotFoundException {
         // pull data from state store of this instance.
         log.info("Pulling stream information from local instance's state-store for streamName={} ; consumerName={}", streamName,
             consumerName);
         Optional<AvroStream> streamValue = kStreams.getAvroStreamForKey(
             AvroStreamKey.newBuilder().setStreamName(streamName).build());
         if (streamValue.isPresent()) {
             streamValue.get().setOperationType(OperationType.GET);
-            for (com.homeaway.digitalplatform.streamregistry.Consumer consumer : streamValue.get().getConsumers()) {
-                if (consumer.getActor().getName().equals(consumerName))
-                    return Optional.of(AvroToJsonDTO.getJsonConsumer(consumer));
+            if (streamValue.get().getConsumers() != null) {
+                for (com.homeaway.digitalplatform.streamregistry.Consumer consumer : streamValue.get().getConsumers()) {
+                    if (consumer.getActor().getName().equals(consumerName))
+                        return Optional.of(AvroToJsonDTO.getJsonConsumer(consumer));
+                }
             }
+        } else {
+            throw new StreamNotFoundException(String.format("StreamName=%s not found. Please create the Stream before getting a Consumer", streamName));
```

Comment:
can we invert this if/else statement?
```
if (!streamValue.isPresent()) {
  throw new StreamNotFoundException(String.format("StreamName=%s not found. Please create the Stream before getting a Consumer", streamName));
}
```

Reply:
@neoword 
fixed it.

Threw the StreamNotFoundException right away at the top of the methods

```registerConsumer
getConsumer
getAllConsumers
deleteConsumers```

```registerProducer
getProducer
getAllProducers
deleteProducer```